upload.py

```diff
@@ -86,6 +116,9 @@ def upload_arguments() -> dict[str, str | None]:
         version = version.strip().lower()
         upload_args['version'] = get_version_from_git() if version == 'git' else version
 
+    if not os.getenv('IDF_COMPONENT_API_TOKEN'):
+        set_oidc_token()
```

Comment:
Current design separates work with the environment variable IDF_COMPONENT_API_TOKEN between this fragment and set_oidc_token.

I suggest making set_oidc_token a pure function

```suggestion
        os.environ['IDF_COMPONENT_API_TOKEN'] = get_oidc_token()
```